pkg/utils/k8sutil.go

```diff
@@ -471,184 +267,94 @@ func GetReadyPod(pods *v1.PodList) (v1.Pod, error) {
+	depName, _ := kubelessRuntime.GetRuntimeDepName(runtime)
+
+	// Prepare Function file and dependencies
+	switch contentType {
+	case "base64":
+		// File is encoded in base64
+		prepareCommand = appendToCommand(prepareCommand, fmt.Sprintf("cat %s | base64 -d > %s", originFile, originFile))
+		break
+	case "text":
+	case "":
+		// Assumming that function is plain text
+		// So we don't need to preprocess it
 	default:
-		return ""
+		return v1.Container{}, fmt.Errorf("Unable to prepare function of type %s: Unknown format", contentType)
 	}
-}
 
-// specify command for the init container
-func getCommand(runtime string, env []v1.EnvVar) []string {
-	switch {
-	case strings.Contains(runtime, "python"):
-		return []string{"pip", "install", "--prefix=/pythonpath", "-r", "/requirements/requirements.txt"}
-	case strings.Contains(runtime, "nodejs"):
-		registry := "https://registry.npmjs.org"
-		scope := ""
-		for _, v := range env {
-			if v.Name == "NPM_REGISTRY" {
-				registry = v.Value
-			}
-			if v.Name == "NPM_SCOPE" {
-				scope = v.Value + ":"
-			}
+	// Validate checksum
+	if checksum == "" {
+		// DEPRECATED: Checksum may be empty
+	} else {
+		checksumInfo := strings.Split(checksum, ":")
+		switch checksumInfo[0] {
+		case "sha256":
+			shaFile := originFile + ".sha256"
+			prepareCommand = appendToCommand(prepareCommand,
+				fmt.Sprintf("echo '%s  %s' > %s", checksumInfo[1], originFile, shaFile),
+				fmt.Sprintf("sha256sum -c %s", shaFile),
+			)
+			break
+		default:
+			return v1.Container{}, fmt.Errorf("Unable to verify checksum %s: Unknown format", checksum)
 		}
-		return []string{"/bin/sh", "-c", "cp package.json /nodepath && npm config set " + scope + "registry " + registry + " && npm install --prefix=/nodepath"}
-	case strings.Contains(runtime, "ruby"):
-		return []string{"bundle", "install", "--path", "/rubypath"}
-	default:
-		return []string{}
 	}
-}
 
-// specify volumes for the init container
-func getVolumeMounts(name, runtime string) []v1.VolumeMount {
-	switch {
-	case strings.Contains(runtime, "python"):
-		return []v1.VolumeMount{
-			{
-				Name:      "pythonpath",
-				MountPath: "/pythonpath",
-			},
-			{
-				Name:      name,
-				MountPath: "/requirements",
-			},
-		}
-	case strings.Contains(runtime, "nodejs"):
-		return []v1.VolumeMount{
-			{
-				Name:      "nodepath",
-				MountPath: "/nodepath",
-			},
-			{
-				Name:      name,
-				MountPath: "/requirements",
-			},
-		}
-	case strings.Contains(runtime, "ruby"):
-		return []v1.VolumeMount{
-			{
-				Name:      "rubypath",
-				MountPath: "/rubypath",
-			},
-			{
-				Name:      name,
-				MountPath: "/requirements",
-			},
-		}
-	case strings.Contains(runtime, "dotnetcore"):
-		return []v1.VolumeMount{
-			{
-				Name:      "dotnetcorepath",
-				MountPath: "/dotnetcorepath",
-			},
-			{
-				Name:      name,
-				MountPath: "/requirements",
-			},
+	// Extract content in case it is a Zip file
+	if filepath.Ext(fileName) == ".zip" {
```

Comment:
I feel like this should be part of the content type rather than intuited from the filename.  In particular, there may be runtimes that _want_ zipfiles directly without having them unpacked (java?) - in which case this would be contentType == "base64" vs "base64+zip"